packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/lambda-walkthrough.ts

```diff
@@ -262,6 +262,8 @@ export async function updateWalkthrough(context, lambdaToUpdate?: string) {
   functionParameters.dependsOn = consolidateDependsOnForLambda(projectMeta, functionParameters.dependsOn, lambdaToUpdate, selectedSettings);
   if (selectedSettings.includes(environmentVariableSetting)) {
     merge(functionParameters, await askEnvironmentVariableQuestions(context, lambdaToUpdate));
+  } else {
+    merge(functionParameters, await askEnvironmentVariableQuestions(context, lambdaToUpdate, undefined, true));
   }
```

Comment:
nit: can replace with:
```suggestion
  merge(functionParameters, await askEnvironmentVariableQuestions(context, lambdaToUpdate, undefined, !selectedSettings.includes(environmentVariableSetting)));
```

Reply:
Thank you! I integrated this into this branch :)